src/index.ts

```diff
@@ -0,0 +1,108 @@
+import OriginalModule = require('module')
+import { dirname, join, resolve } from 'path'
+import * as callerId from 'caller-id'
+
+const Module = OriginalModule as typeof OriginalModule & {
+  _load: (request: string, parent: NodeModule) => NodeModule
+  globalPaths: string[]
+  _resolveFilename: (filename: string) => string
+}
```

Comment:
It seems that `@types/node` is incomplete in this regard so I had to improvise here.